django/contrib/admin/filters.py

```diff
@@ -273,8 +281,10 @@ def expected_parameters(self):
 
     def choices(self, changelist):
         yield {
-            'selected': self.lookup_val is None,
-            'query_string': changelist.get_query_string({}, [self.lookup_kwarg]),
+            'selected': self.lookup_val is None or self.lookup_val == self.sentinel,
+            'query_string': changelist.get_query_string({
+                self.lookup_kwarg: self.sentinel
```

Comment:
Wow, thanks for the feedback, I'll have a look tomorrow to fix all that!

On Thu, 10 Mar 2016, 18:52 Tim Graham, <notifications@github.com> wrote:

> In django/contrib/admin/filters.py
> <https://github.com/django/django/pull/5579#discussion_r55719801>:
>
> > @@ -273,8 +281,10 @@ def expected_parameters(self):
> >
> >      def choices(self, changelist):
> >          yield {
> > -            'selected': self.lookup_val is None,
> > -            'query_string': changelist.get_query_string({}, [self.lookup_kwarg]),
> > +            'selected': self.lookup_val is None or self.lookup_val == self.sentinel,
> > +            'query_string': changelist.get_query_string({
> > +                self.lookup_kwarg: self.sentinel
>
> adding trailing comma
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/django/django/pull/5579/files#r55719801>.
>